test/dynamo/test_subclasses.py

```diff
@@ -128,6 +143,74 @@ def f(x):
         self.assertEqual(cnt.frame_count, 3)
         self.assertEqual(cnt.op_count, 3)
 
+    def test_compile_with_functionalization(self):
+        x = torch.randn([3, 4])
+        x_clone = x.clone()
+        x_clone2 = x.clone()
+        backend = EagerRecordGraphAndInputs()
+        cnt = torch._dynamo.testing.CompileCounterWithBackend(backend)
+
+        @torch.compile(backend=cnt, fullgraph=True)
+        def f(x):
+            return x.add_(1.0) + torch.nn.functional.relu_(x)
+
+        f_out = f(x)
+        self.assertEqual(cnt.frame_count, 1)
+        self.assertEqual(cnt.op_count, 3)
+        self.assertEqual(len(backend.graphs), 1)
+        self.assertEqual(len(backend.example_inputs), 1)
+
+        expected = """\
+class GraphModule(torch.nn.Module):
+    def forward(self, L_x_ : torch.Tensor):
+        l_x_ = L_x_
+
+        add_ = l_x_.add_(1.0)
+        relu_ = torch.relu_(l_x_);  l_x_ = None
+        add = add_ + relu_;  add_ = relu_ = None
+        return (add,)
+"""
+        actual = normalize_gm(backend.graphs[0].print_readable(print_output=False))
+        self.assertEqual(actual, expected)
+
+        ff = torch.func.functionalize(f)
+        ff_out = ff(x_clone)
+
+        self.assertEqual(cnt.frame_count, 2)
+        self.assertEqual(cnt.op_count, 6)
+        self.assertEqual(len(backend.graphs), 2)
+        self.assertEqual(len(backend.example_inputs), 2)
+        actual = normalize_gm(backend.graphs[1].print_readable(print_output=False))
```

Comment:
I'd recommend not checking cnt and backend again, because we already did it one time? Ditto below

Reply:
Gotcha, thanks for the explanation. I'm convinced, let's leave them